src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,149 @@
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void existingLogin_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        storageDao.add(user);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerShortLogin_NotOk() {
+        User user = createNewUser(SHORT_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void emptyLogin_NotOk() {
+        User user = createNewUser("", VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerShortPassword_NotOk() {
+        User user = createNewUser(VALID_LOGIN, SHORT_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void passwordNullValue_NotOk() {
+        User user = createNewUser(VALID_LOGIN, null, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void emptyPassword_NotOk() {
+        User user = createNewUser(VALID_LOGIN, "", VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void ageNullValue_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void underAgeUser_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, UNDERAGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void negativeUserAge_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, NEGATIVE_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void zeroAgeValue_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, ZERO_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void largeAge_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, OVER_AGE);
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals(LARGE_AGE_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    void addNewUser_Ok() {
```

Comment:
This test method `addNewUser_Ok` should be split into two: one to check if the user was added to storage, and another to check if the correct user is returned by the `register` method.